docs/apache-airflow/authoring-and-scheduling/datasets.rst

```diff
@@ -63,28 +63,46 @@ A dataset is defined by a Uniform Resource Identifier (URI):
 
 Airflow makes no assumptions about the content or location of the data represented by the URI. It is treated as a string, so any use of regular expressions (eg ``input_\d+.csv``) or file glob patterns (eg ``input_2022*.csv``) as an attempt to create multiple datasets from one declaration will not work.
 
-There are two restrictions on the dataset URI:
+A dataset should be created with a valid URI. Airflow core and providers define various URI schemes that you can use, such as ``file`` (core), ``https`` (by the HTTP provider), and ``s3`` (by the Amazon provider). Third-party providers and plugins may also provide their own schemes. These pre-defined schemes have individual semantics that are expected to be followed.
```

Comment:
You mention the https provider defines a scheme, but I don't see that in this PR?

Reply:
I intended to add some more but decided to hold back some of those after the PR got long. I’ll reword this.